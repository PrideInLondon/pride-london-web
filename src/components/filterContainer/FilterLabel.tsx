import React from 'react'
import { Label, Text, Input } from './FilterLabel.styles'
import { FilterLabelProps } from './FilterLabel.types'

const FilterLabel: React.FC<FilterLabelProps> = ({
  filterName,
  filterColour,
  filterType,
  isSelected,
  handleSelect,
}) => (
  <Label
    filterColour={filterColour}
    isSelected={isSelected}
    htmlFor={filterName}
    tabIndex={0}
  >
    <Text>{filterName}</Text>
    <Input
      onClick={() => handleSelect(filterName)}
      type={filterType}
      value={filterName}
      id={filterName}
      name="filterLabel"
      tabIndex={-1}
      aria-label={filterName}
    />
  </Label>
)

export default FilterLabel
