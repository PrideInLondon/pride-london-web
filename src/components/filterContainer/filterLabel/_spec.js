import React from 'react'
import { mount, shallow } from 'enzyme'
import FilterLabel from '.'

describe('FilterLabel', () => {
  it('renders when selected', () => {
    const wrapper = mount(
      <FilterLabel
        filterName="foo"
        filterColour="bar"
        filterType="checkbox"
        isSelected
        handleSelect={() => {}}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders when not selected', () => {
    const wrapper = mount(
      <FilterLabel
        filterName="foo"
        filterColour="bar"
        filterType="radio"
        isSelected={false}
        handleSelect={() => {}}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('executes handler when filter is clicked', () => {
    const handleSelect = jest.fn()
    const wrapper = shallow(
      <FilterLabel
        filterName="foo"
        filterColour="bar"
        filterType="checkbox"
        isSelected
        handleSelect={handleSelect}
      />
    )
    wrapper.find('styles__Input').simulate('click')
    expect(handleSelect).toHaveBeenCalledTimes(1)
    expect(handleSelect).toHaveBeenCalledWith('foo')
  })
})
