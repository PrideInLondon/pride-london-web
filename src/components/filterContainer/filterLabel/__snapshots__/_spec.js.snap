// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterLabel renders when not selected 1`] = `
<FilterLabel
  filterColour="bar"
  filterName="foo"
  filterType="radio"
  handleSelect={[Function]}
  isSelected={false}
>
  <styles__Label
    backgroundColor="bar"
    htmlFor="foo"
    isSelected={false}
    tabIndex="0"
  >
    <StyledComponent
      backgroundColor="bar"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "styles__Label-sc-1wu4f6z-0",
            "isStatic": false,
            "lastClassName": "fekPyo",
            "rules": Array [
              "align-items:center;border:2px solid;background-color:",
              [Function],
              ";border-color:",
              [Function],
              ";border-radius:4px;color:",
              [Function],
              ";cursor:pointer;display:flex;font-family:",
              "Poppins, sans-serif",
              ";font-size:1rem;height:32px;justify-content:center;padding:0 20px;margin-right:10px;margin-bottom:10px;white-space:nowrap;",
            ],
          },
          "displayName": "styles__Label",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "styles__Label-sc-1wu4f6z-0",
          "target": "label",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      htmlFor="foo"
      isSelected={false}
      tabIndex="0"
    >
      <label
        className="styles__Label-sc-1wu4f6z-0 fekPyo"
        htmlFor="foo"
        tabIndex="0"
      >
        <styles__Text>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "styles__Text-sc-1wu4f6z-1",
                  "isStatic": true,
                  "lastClassName": "cywyfl",
                  "rules": Array [],
                },
                "displayName": "styles__Text",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "styles__Text-sc-1wu4f6z-1",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <span
              className="styles__Text-sc-1wu4f6z-1 cywyfl"
            >
              foo
            </span>
          </StyledComponent>
        </styles__Text>
        <styles__Input
          aria-label="foo"
          id="foo"
          name="filterLabel"
          onClick={[Function]}
          tabIndex="-1"
          type="radio"
          value="foo"
        >
          <StyledComponent
            aria-label="foo"
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [
                  [Function],
                ],
                "componentStyle": ComponentStyle {
                  "componentId": "styles__Input-sc-1wu4f6z-2",
                  "isStatic": false,
                  "lastClassName": "kgGtRH",
                  "rules": Array [
                    "border: 0;",
                    "clip: rect(0 0 0 0);",
                    "clip-path: inset(50%);",
                    "height: 1px;",
                    "margin: -1px;",
                    "overflow: hidden;",
                    "padding: 0;",
                    "position: absolute;",
                    "white-space: nowrap;",
                    "width: 1px;",
                    ";",
                  ],
                },
                "displayName": "styles__Input",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "styles__Input-sc-1wu4f6z-2",
                "target": "input",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            id="foo"
            name="filterLabel"
            onClick={[Function]}
            tabIndex="-1"
            type="radio"
            value="foo"
          >
            <input
              aria-label="foo"
              className="styles__Input-sc-1wu4f6z-2 kgGtRH"
              id="foo"
              name="filterLabel"
              onClick={[Function]}
              tabIndex="-1"
              type="radio"
              value="foo"
            />
          </StyledComponent>
        </styles__Input>
      </label>
    </StyledComponent>
  </styles__Label>
</FilterLabel>
`;

exports[`FilterLabel renders when selected 1`] = `
<FilterLabel
  filterColour="bar"
  filterName="foo"
  filterType="checkbox"
  handleSelect={[Function]}
  isSelected={true}
>
  <styles__Label
    backgroundColor="bar"
    htmlFor="foo"
    isSelected={true}
    tabIndex="0"
  >
    <StyledComponent
      backgroundColor="bar"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "styles__Label-sc-1wu4f6z-0",
            "isStatic": false,
            "lastClassName": "jsCQny",
            "rules": Array [
              "align-items:center;border:2px solid;background-color:",
              [Function],
              ";border-color:",
              [Function],
              ";border-radius:4px;color:",
              [Function],
              ";cursor:pointer;display:flex;font-family:",
              "Poppins, sans-serif",
              ";font-size:1rem;height:32px;justify-content:center;padding:0 20px;margin-right:10px;margin-bottom:10px;white-space:nowrap;",
            ],
          },
          "displayName": "styles__Label",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "styles__Label-sc-1wu4f6z-0",
          "target": "label",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      htmlFor="foo"
      isSelected={true}
      tabIndex="0"
    >
      <label
        className="styles__Label-sc-1wu4f6z-0 jsCQny"
        htmlFor="foo"
        tabIndex="0"
      >
        <styles__Text>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "styles__Text-sc-1wu4f6z-1",
                  "isStatic": true,
                  "lastClassName": "cywyfl",
                  "rules": Array [],
                },
                "displayName": "styles__Text",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "styles__Text-sc-1wu4f6z-1",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <span
              className="styles__Text-sc-1wu4f6z-1 cywyfl"
            >
              foo
            </span>
          </StyledComponent>
        </styles__Text>
        <styles__Input
          aria-label="foo"
          id="foo"
          name="filterLabel"
          onClick={[Function]}
          tabIndex="-1"
          type="checkbox"
          value="foo"
        >
          <StyledComponent
            aria-label="foo"
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [
                  [Function],
                ],
                "componentStyle": ComponentStyle {
                  "componentId": "styles__Input-sc-1wu4f6z-2",
                  "isStatic": false,
                  "lastClassName": "kgGtRH",
                  "rules": Array [
                    "border: 0;",
                    "clip: rect(0 0 0 0);",
                    "clip-path: inset(50%);",
                    "height: 1px;",
                    "margin: -1px;",
                    "overflow: hidden;",
                    "padding: 0;",
                    "position: absolute;",
                    "white-space: nowrap;",
                    "width: 1px;",
                    ";",
                  ],
                },
                "displayName": "styles__Input",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "styles__Input-sc-1wu4f6z-2",
                "target": "input",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            id="foo"
            name="filterLabel"
            onClick={[Function]}
            tabIndex="-1"
            type="checkbox"
            value="foo"
          >
            <input
              aria-label="foo"
              className="styles__Input-sc-1wu4f6z-2 kgGtRH"
              id="foo"
              name="filterLabel"
              onClick={[Function]}
              tabIndex="-1"
              type="checkbox"
              value="foo"
            />
          </StyledComponent>
        </styles__Input>
      </label>
    </StyledComponent>
  </styles__Label>
</FilterLabel>
`;
